nexus_constructor/geometry/disk_chopper_geometry.py

```diff
@@ -0,0 +1,497 @@
+from math import pi
+
+from PySide2.QtGui import QVector3D
+from PySide2.QtWidgets import QListWidget
+import numpy as np
+from numpy import diff, unique
+from numpy.core.umath import deg2rad, ndarray
+
+from nexus_constructor.geometry import OFFGeometryNoNexus
+from nexus_constructor.validators import DATASET_TYPE
+
+SLIT_EDGES = "slit_edges"
+SLITS = "slits"
+RADIUS = "radius"
+SLIT_HEIGHT = "slit_height"
+
+REQUIRED_CHOPPER_FIELDS = [SLIT_EDGES, SLITS, RADIUS, SLIT_HEIGHT]
+CONTAINS_INTEGER = [
+    DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Integer" in key
+]
+CONTAINS_SHORT = [DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Short" in key]
+CONTAINS_LONG = [DATASET_TYPE[key] for key in DATASET_TYPE.keys() if "Long" in key]
+INT_TYPES = CONTAINS_INTEGER + CONTAINS_SHORT + CONTAINS_LONG
+
+FLOAT_TYPES = [
+    DATASET_TYPE[key] for key in DATASET_TYPE.keys() if key in ["Float", "Double"]
+]
+
+
+class ChopperChecker:
+    def __init__(self, fields_widget: QListWidget):
+
+        self.fields_widget = fields_widget
+        self.fields_dict = dict()
+
+        for i in range(fields_widget.count()):
+            widget = fields_widget.itemWidget(fields_widget.item(i))
+            self.fields_dict[widget.name] = widget
+
+        self._slits = None
+        self._slit_edges = None
+        self._radius = None
+        self._slit_height = None
+        self._units = "deg"
+
+    def validate_chopper(self):
+
+        return (
+            self.required_fields_present()
+            and self.fields_have_correct_type()
+            and self.edges_array_has_correct_shape()
+            and self.input_describes_valid_chopper()
+        )
+
+    def required_fields_present(self):
+        """
+        Carries out a preliminary check of the fields input to see if seems like it might be describing a
+        valid disk chopper shape.
+        """
+        return set(self.fields_dict.keys()) == set(REQUIRED_CHOPPER_FIELDS)
+
+    def fields_have_correct_type(self):
+
+        return (
+            self.fields_dict[SLITS].dtype in INT_TYPES
+            and self.fields_dict[RADIUS].dtype in FLOAT_TYPES
+            and self.fields_dict[SLIT_HEIGHT].dtype in FLOAT_TYPES
+            and self.fields_dict[SLIT_EDGES].dtype in FLOAT_TYPES
+        )
+
+    def edges_array_has_correct_shape(self):
+
+        edges_dim = self.fields_dict[SLIT_EDGES].value.ndim
+
+        if edges_dim > 2:
+            return False
+
+        if edges_dim == 2:
+            edges_shape = self.fields_dict[SLIT_EDGES].value.shape
+            if edges_shape[0] != 1 and edges_shape[1] != 1:
+                return False
+
+        return True
+
+    def input_describes_valid_chopper(self):
+
+        self._slit_edges = self.fields_dict[SLIT_EDGES].value
+        self._radius = self.fields_dict[RADIUS].value
+        self._slit_height = self.fields_dict[SLIT_HEIGHT].value
+        self._slits = self.fields_dict[SLITS].value
```

Comment:
Could you construct a `ChopperDetails` at this point instead of duplicating these data as both members of `ChopperChecker ` and `ChopperDetails`?